src/bin/console/console.cpp

```diff
@@ -539,40 +467,27 @@ int Console::visualize(const std::string& input) {
       LQPVisualizer visualizer;
       visualizer.visualize(lqp_roots, graph_filename, img_filename);
     } catch (const std::exception& exception) {
-      out("Exception while creating query plan:\n  " + std::string(exception.what()) + "\n");
+      out(std::string(exception.what()) + "\n");
       return ReturnCode::Error;
     }
   } else {
     // Compile the parse result
+    SQLQueryPlan plan;
     try {
-      plan = SQLPlanner::plan(parse_result);
+      plan = _sql_pipeline->get_query_plan();
     } catch (const std::exception& exception) {
-      out("Exception thrown while compiling query plan:\n  " + std::string(exception.what()) + "\n");
+      out(std::string(exception.what()) + "\n");
       return ReturnCode::Error;
     }
 
     if (mode != "noexec") {
-      const auto auto_commit = (_explicitly_created_transaction_context == nullptr);
-      auto transaction_context =
-          auto_commit ? TransactionManager::get().new_transaction_context() : _explicitly_created_transaction_context;
-
-      plan.set_transaction_context(transaction_context);
-
-      if (_execute_plan(plan) == ReturnCode::Error) {
-        transaction_context->rollback();
+      try {
+        _sql_pipeline->get_result_table();
+      } catch (const std::exception& exception) {
+        out("An operator failed and the transaction has been rolled back:\n" + std::string(exception.what()) + "\n");
```

Comment:
this could be any error, right? Parsing, Assert, Whatnot... And rollback only happens when a MVCC action fails, right? So this error message night be misleading

Reply:
No, if an operator "crashes" (e.g., wrong data types), we also rollback the transaction.